### DISCUSSION TITLE
Anomaly Detectors could have a threshold method?

### DISCUSSION THREAD

**Discussion by yuritpinheiro** [online-ml/river#614] [General]
The base class have only three methods: `clone`, `learn_one`, `score_one`. I would like to implement a `AnomalyDetector` with it specified threshold and this would lead to creating a method for the current threshold or to evaluate one sample and provide a boolean response (is a anomaly or is not a anomaly).

**Comment by MaxHalford** [comment]
Hey! Thanks for the question. What prevents you from just taking the output of `score_one(x)` and thresholding it manually? What other logic would this `threshold` method contain?

**Reply by yuritpinheiro** [reply]
I thought about this and I think what I need is an attribute to indicating that an anomaly has been detected. In fact, it would make the `AnomalyDetector` more similar to the `DriftDetector`.

Answering your question directly, nothing prevents me from doing it although the threshold of this method depends upon internal parameters, for example, the number of samples processed. The method for calculating the threshold would use the internal parameters and continuously update the threshold.

Finally, I think that I will use a private method to determine the threshold and use a public attribute to expose the detection. What do you think?

**Reply by MaxHalford** [reply]
Hello there, sorry for the late answer. I'm not against having logic to classify rows as anomalies. It could be some higher level construct that would classify an observation as an anomaly when the score is in the x% of highest scores, or something like that. I don't have a usecase for it at the moment, so I won't work on it, but feel free to propose something.
